lib/ansible/modules/cloud/google/gce_facts.py

```diff
@@ -0,0 +1,359 @@
+    type: string
+    sample: test-network
+zone:
+    description: Name of the zone where instance resides.
+    returned: always
+    type: string
+    sample: us-central1-a
+private_ip:
+    description: Private ip of instance.
+    returned: always
+    type: string
+    sample: 10.115.0.2
+public_ip:
+    description: Public ip of instance.
+    returned: always
+    type: string
+    sample: 35.224.101.234
+status:
+    description: State of the instance.
+    returned: always
+    type: string
+    sample: RUNNING
+disks:
+    description: Disks including boot disk which are attached to instance.
+    returned: always
+    type: list
+    sample: ["test-disk-000", "test-disk-001"]
+image:
+    description: Image used by instance.
+    returned: always
+    type: string
+    sample: centos-7
+machine_type:
+    description: Type of machine.
+    returned: always
+    type: string
+    sample: n1-standard-1
+metadata:
+    description: Metadata of instance.
+    returned: always
+    type: dictionary
+    sample: {'Type':'Server', 'App':'Nginx'}
+tags:
+    description: Tags of instance.
+    returned: always
+    type: list
+    sample: ['global', 'test']
+'''
+
+try:
+    import libcloud
+    from libcloud.compute.types import Provider
+    gce_provider = Provider.GCE
+    HAS_LIBCLOUD = True
+except ImportError:
+    HAS_LIBCLOUD = False
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.gce import gce_connect, unexpected_error_msg
+from ansible.module_utils.gcp import get_valid_location
+
+
+def get_instance_info(inst):
+    """Retrieves instance information from an instance object and returns it
+    as a dictionary.
+
+    """
+    metadata = {}
+    if 'metadata' in inst.extra and 'items' in inst.extra['metadata']:
+        for md in inst.extra['metadata']['items']:
+            metadata[md['key']] = md['value']
+
+    try:
+        netname = inst.extra['networkInterfaces'][0]['network'].split('/')[-1]
+    except:
+        netname = None
+    try:
+        subnetname = inst.extra['networkInterfaces'][0]['subnetwork'].split('/')[-1]
+    except:
+        subnetname = None
+    if 'disks' in inst.extra:
+        disk_names = [disk_info['source'].split('/')[-1]
+                      for disk_info
+                      in sorted(inst.extra['disks'],
+                                key=lambda disk_info: disk_info['index'])]
+    else:
+        disk_names = []
+
+    if len(inst.public_ips) == 0:
+        public_ip = None
+    else:
+        public_ip = inst.public_ips[0]
+
+    return({
+        'image': inst.image is not None and inst.image.split('/')[-1] or None,
+        'disks': disk_names,
+        'machine_type': inst.size,
+        'metadata': metadata,
+        'name': inst.name,
+        'network': netname,
+        'subnetwork': subnetname,
+        'private_ip': inst.private_ips[0],
+        'public_ip': public_ip,
+        'status': ('status' in inst.extra) and inst.extra['status'] or None,
```

Comment:
Please separate this clause into separate lines for maximum readability (like above).